public/bitemp_graph_helper.js

```diff
@@ -144,8 +142,88 @@ function edit(uri) {
   }
 }
 
-function deleteDoc(uri) {
-  if (!uri) { // Not given a valid document uri
+function getDocColl(uri) {
+  console.log('Getting docColl');
+  var docColl = $.ajax({
+    url: "/v1/documents?uri="+uri+"&category=collections&format=json",
+    success: function(data, textStatus) {
+     console.log('got collections: ' + data);
+    },
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log('problem');
+    },
+    async: false,
+  });
+ 
+ return JSON.parse(docColl.responseText);
+}
+
+var deleteDoc = function (chart) {
+  var uri = chart.getLogicalURI();
+  var ajax = true;
+  if (!uri) {
+    return;
+  }
+  var collection = getDocColl(uri);
+  var url = '/v1/documents?uri=' + uri + '&temporal-collection='+collection.collections[0];
+  var currDate = new Date();
+  
+  $.ajax(
+  {
+    url: 'http://localhost:3000/v1/resources/temporal-range?rs:collection='+collection.collections[0],
+    success: function(response, textStatus)
+    {
+      succFunc(response);
+    },
+    error: function(jqXHR, textStatus, errorThrown)
+    {
+      console.log('problem');
+    }
+  });
+  
+  var succFunc = function(response) {
+    var sysBoxDate;
+    var tempDate = new Date(response.sysEnd);
+    
+    //Add a system time to ajax request if specified
+    if (document.getElementById('sysStartBox').value !== '') {
+      url += '&system-time='+document.getElementById('sysStartBox').value;
+      sysBoxDate = new Date(document.getElementById('sysStartBox').value);
+    } 
+    
+    if (!sysBoxDate) {
+      if (tempDate.valueOf() > currDate.valueOf()) {
+        ajax = false;
+        document.getElementById('deleteErrMessage').innerHTML = 'Error: System time does not go backward.'.bold() + ' Current time is ' + tempDate;
+      }
+    }
+    else if (tempDate.valueOf() > sysBoxDate.valueOf()) {
+      document.getElementById('deleteErrMessage').innerHTML = 'Error: System time does not go backward.'.bold() + ' Current time is ' + tempDate;
+      ajax = false;
+    }
+    
```

Comment:
can be simplified to 

var sysBoxDate= document.getElementById('sysStartBox').value;
if (!sysBoxDate) { url+= ..; } else sysBoxDate=currentData;
if (tempDate.valueOf() > sysBoxDate.valueOf()){
document.getElementById('deleteErrMessage').innerHTML = 'Error: System time does not go backward.'.bold() + ' Current time is ' + tempDate;
ajax=false;
}